File: GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/encoder/BitLength.java
package com.gdpr.phase2.encoder;


import java.util.HashMap;
import java.util.Map;

public class BitLength {

    public static Map<String,Integer> fieldLengths = new HashMap<String, Integer>(){{
        put("version",6);
        put("anyBoolean",1);
        put("singleOrRange",1);
        put("encodingType",1);
        put("checksum",18);
        put("created",36);
        put("lastUpdated",36);
        put("cmpId",12);
        put("cmpVersion",12);
        put("consentScreen",6);
        put("consentLanguage",12);
        put("vendorListVersion",12);
        put("policyVersion",6);
        put("isServiceSpecific",1);
        put("useNonStandardStacks",1);
        put("purposeOneTreatment",1);
        put("publisherCountryCode",12);
        put("specialFeatureOptIns",12);
        put("purposeConsents",24);
        put("purposeLegitimateInterest",24);
        put("vendorId",16);
        put("purposeId",6);
        put("numEntries",12);
        put("maxId",16);
        put("restrictionType",2);
        put("numRestrictions",12);
        put("segmentType",3);
        put("publisherPurposeConsents",24);
        put("publisherLegitimateInterest",24);
        put("numCustomPurposes",6);
    }};
}

File: modules/core/src/encoder/BitLength.ts
import {Fields} from '../model/index.js';

export class BitLength {

  public static readonly [Fields.cmpId]: number = 12;
  public static readonly [Fields.cmpVersion]: number = 12;
  public static readonly [Fields.consentLanguage]: number = 12;
  public static readonly [Fields.consentScreen]: number = 6;
  public static readonly [Fields.created]: number = 36;
  public static readonly [Fields.isServiceSpecific]: number = 1;
  public static readonly [Fields.lastUpdated]: number = 36;
  public static readonly [Fields.policyVersion]: number = 6;
  public static readonly [Fields.publisherCountryCode]: number = 12;
  public static readonly [Fields.publisherLegitimateInterests]: number = 24;
  public static readonly [Fields.publisherConsents]: number = 24;
  public static readonly [Fields.purposeConsents]: number = 24;
  public static readonly [Fields.purposeLegitimateInterests]: number = 24;
  public static readonly [Fields.purposeOneTreatment]: number = 1;
  public static readonly [Fields.specialFeatureOptins]: number = 12;
  public static readonly [Fields.useNonStandardTexts]: number = 1;
  public static readonly [Fields.vendorListVersion]: number = 12;
  public static readonly [Fields.version]: number = 6;

  public static readonly anyBoolean: number = 1;
  public static readonly encodingType: number = 1;
  public static readonly maxId: number = 16;
  public static readonly numCustomPurposes: number = 6;
  public static readonly numEntries: number = 12;
  public static readonly numRestrictions: number = 12;
  public static readonly purposeId: number = 6;
  public static readonly restrictionType: number = 2;
  public static readonly segmentType: number = 3;
  public static readonly singleOrRange: number = 1;
  public static readonly vendorId: number = 16;

}
